modules/extension/wmts/src/main/java/org/geotools/ows/wmts/model/WMTSCapabilities.java

```diff
@@ -330,20 +330,20 @@ private TileMatrixSet parseMatrixSet(TileMatrixSetType tm)
     }
 
     private WMTSLayer parseLayer(LayerType layerType) {
-        String title = ((LanguageStringType) layerType.getTitle().get(0)).getValue();
+
+        String name = layerType.getIdentifier().getValue();
+        String title =
+                layerType.getTitle().size() > 0
+                        ? ((LanguageStringType) layerType.getTitle().get(0)).getValue()
+                        : name;
+
         WMTSLayer layer = new WMTSLayer(title);
-        layer.setName(layerType.getIdentifier().getValue());
+        layer.setName(name);
 
-        // The Abstract is of Type LanguageStringType, not String.
-        StringBuilder sb = new StringBuilder();
-        for (Object line : layerType.getAbstract()) {
-            if (line instanceof LanguageStringType) {
-                sb.append(((LanguageStringType) line).getValue());
-            } else {
-                sb.append(line);
-            }
-        } // end of for
-        layer.set_abstract(sb.toString());
+        // The Abstract is of Type LanguageStringType, not String. We're choosing the first one.
+        if (layerType.getAbstract().size() > 0) {
+            layer.set_abstract(((LanguageStringType) layerType.getAbstract().get(0)).getValue());
```

Comment:
I know the problem from WFS. The starting point here is the constructor that takes net.opengis.wmts.v_1.CapabilitiesType as argument. Wouldn't that suggest that the whole response are correct, and there is no need for any further type checking?

Reply:
By personal experience I cannot assume that, would need to check the actual code... the Parser has been designed to be lenient and has a tendency to return maps/strings when there is a mismatch between the actual elements and the ones the schema wants. In other words, the current code could be parsing invalid capabilities documents stiching toghether what it finds by force of toString, while the new code would not any longer.

Reply:
I've added a test with an abstract having html-element. Which I think some might add. Then I reverted to the old handling of abstract's. The result is that everything outside of the html-element is returned.

What I was afraid of with the old approach is that it will join abstracts from different languages into one.

Reply:
Maybe a middle ground? Loop over, but stop at the first in case it's a language string (sane case?) otherwise keep on looping. Basically the original code, plus a "break"?

Reply:
Then I check for LanguageStringType and break at first occurrence.